ls1intum/Artemis
# Refactor the onError utils method to use the correct translations

### Checklist

### Motivation and Context

We have a lot of "translation not found" errors on sentry with translation keys which are actually http error messages.

### Description

We display the correct translation for those, and the error message directly if we don't have a translation. We use the shared function for that to reduce duplication.

### Steps for Testing

I'd suggest verifying the changes of this PR by looking at the actual code changes. It should be relatively verifiable that functionality wise nothing is broken. But you can also test if error messages are still displayed, like:
![grafik](https://user-images.githubusercontent.com/72132281/122775622-9de16600-d2aa-11eb-9498-2949587ee7f1.png)

You can provoke errors by manually enabling disabled buttons only browser side, by deleting `disabled=""` from the element in the DOM explorer:
![grafik](https://user-images.githubusercontent.com/72132281/122775697-b18ccc80-d2aa-11eb-8384-72d56ca791f6.png)
->
![grafik](https://user-images.githubusercontent.com/72132281/122775706-b3569000-d2aa-11eb-8575-be44505c61b8.png)

That would allow you to e.g. submit a modelling exercise without (max) scores set which causes the server to respond with an error.
